api/demand.php

```diff
@@ -0,0 +1,120 @@
+
+    $emailId = filter_input(INPUT_GET, 'emailId', FILTER_VALIDATE_INT);
+
+    $userData = FlexAPI::superAccess()->read('userdata', [
+        'filter' => [ 'user' => FlexAPI::guard()->getUsername() ],
+        'flatten' => 'singleResult'
+    ]);
+
+    $demandEmail = FlexAPI::superAccess()->read('email', [
+        'filter' => [
+            'id' => $emailId,
+            'person' => $userData['id']
+        ],
+        'references' => ['format' => 'data'],
+        'flatten' => 'singleResult'
+    ]);
+    if (!$demandEmail) {
+        throw(new Exception('Keine Forderungs-Mail gefunden.', 400));
+    }
+
+    $sonderAktion = FlexAPI::get('sonderAktion');
+    if (time() < $sonderAktion['expires']) {
+        $password = filter_input(INPUT_GET, 'sonderAktion');
+        if ($password === $sonderAktion['password']) {
+            $institution = FlexAPI::superAccess()->read('institution', [
+                'filter' => [ 'id' => $demandEmail['demanded_street_section']['institution'] ],
+                'flatten' => 'singleResult'
+            ]);
+            if (in_array($institution['type'], [3, 4])) {
+                throw(new Exception('Sonderaktion gilt leider nicht für Krankenhäuser und Pflegeheime.', 400));
+            }
+        } else {
+            throw(new Exception('Falsches Aktions-Passwort.', 400));
+        }
+    }
+
+    if ($demandEmail['demanded_street_section']['mail_sent']) {
+        throw(new Exception('Für den Straßenabschnitt wurde bereits eine Forderungsmail verschickt.', 400));
+    }
+
+    $config = include(__DIR__."/../api.conf.php");
+    $dbConnection = new PdoPreparedConnection($config['databaseCredentials']['data']);
+
+    $statement = $dbConnection->executeQuery(""
+        . "SELECT policedepartment.email FROM policedepartment"
+        . " JOIN institution ON ST_CONTAINS(policedepartment.geom, institution.position)"
+        . " WHERE institution.id = :institution",
+        [ 'institution' => $demandEmail['demanded_street_section']['institution'] ]
+    );
+    $policeDepartment = $statement->fetchAll();
+    if (!count($policeDepartment)) {
+        throw(new Exception('Es konnte kein zuständiges Polizeikommissariat gefunden werden.', 400));
+    }
+    $policeDepartment = $policeDepartment[0];
+    $pdEmail = $policeDepartment['email'];
+    // zum TEST:
+    $pdEmail = str_replace('polizei.hamburg.de', 'sven.anders.hamburg', $pdEmail);
+
+    $demandMessage = implode("\n\n", [
+        $demandEmail['mail_start'],
+        $demandEmail['mail_body'],
+        $demandEmail['mail_end']
+    ]);
+
+    $response['message'] = 'Send demand to: '.$pdEmail;
+
+    $mailService = new SmtpMailService($config['mailing']['smtp'], [
+        'address' => $config['defaultFrom']['address'],
+        'name' => $userData['first_name']." ".$userData['last_name']
+    ]);
+    $mailService->send(
+        $pdEmail,
+        $demandEmail['mail_subject'],
+        nl2br($demandMessage),   // HTML
```

Comment:
Oder HTML weglassen?

Reply:
Schadet ja auch nicht, oder?

Reply:
Doch. Wenn ich z.B. in den Text etwas eintrage, was man als HTML verstehen könnte, dann schon.

Also wenn ich zum Beispiel schreibe:

```
Zwischen Straße1 <- Eisenbahnbrücke -> Straße2 muss Tempo 30
```

wird daraus ggf. (nicht getestet) in der HTML Version

```
Zwischen Straße1 Straße2 muss Tempo 30
```

Eine HTML Version belegt zudem Speicherplatz und Resourcen und ist unötig und schlecht fürs Klima. ;-)

Reply:
:D Klima, na gut!

Reply:
Ist auch erledigt